components/pagination/pagination.ts

```diff
@@ -0,0 +1,166 @@
+    {{ p }}
+</a>
+<a *ngIf="hasNavigation" class="item" (click)="setPage(page+1)" [class.disabled]="!hasNext()">
+    <span><i class="angle right icon"></i></span>
+</a>
+<a *ngIf="hasBoundaryLinks" class="item"  (click)="setPage(pageCount)" [class.disabled]="page===pageCount">
+    <span><i class="angle double right icon"></i></span>
+</a>
+`
+})
+export class SuiPagination implements OnChanges {
+
+    @HostBinding("class.ui")
+    @HostBinding("class.pagination")
+    @HostBinding("class.menu")
+    private _paginationClasses:boolean;
+
+    // Public members
+    public pageCount:number;
+
+    @Output()
+    public pageChange:EventEmitter<number>;
+
+    // Private members
+    private _maxSize?:number;
+    private _collectionSize:number;
+    private _pageSize:number;
+    private _page:number;
+    private _hasBoundaryLinks:boolean;
+    private _hasNavigation:boolean;
+
+    @Input()
+    public get maxSize():number {
+        return this._maxSize;
+    }
+
+    public set maxSize(value:number) {
+        this._maxSize = (value !== undefined) ? Math.max(value, 1) : undefined;
+    }
+
+    @Input()
+    public get collectionSize():number {
+        return this._collectionSize;
+    }
+
+    public set collectionSize(value:number) {
+        this._collectionSize = Math.max(value, 0);
+    }
+
+    @Input()
+    public get pageSize():number {
+        return this._pageSize;
+    }
+
+    public set pageSize(value:number) {
+        this._pageSize = value;
+    }
+
+    @Input()
+    public get hasNavigation():boolean {
+        return this._hasNavigation || this._maxSize < this.pageCount;
+    }
+
+    public set hasNavigation(value:boolean) {
+        this._hasNavigation = value;
+    }
+
+    @Input()
+    public get hasBoundaryLinks():boolean {
+        return this._hasBoundaryLinks;
+    }
+
+    public set hasBoundaryLinks(value:boolean) {
+        this._hasBoundaryLinks = value;
+    }
+
+    @Input()
+    public get page():number {
+        return this._page;
+    }
+
+    public set page(value:number) {
+        this.setPage(value);
+    }
+
+    public get pages():number[] {
+        const [start, end] = this.applyPagination();
+
+        return Array<number>(end - start)
+            .fill(start + 1)
+            .map((s, i) => s + i);
+    }
+
+    constructor() {
+        this._paginationClasses = true;
+        this.pageChange = new EventEmitter<number>();
+
+        this.collectionSize = 100;
+        this.pageSize = 10;
+        this.page = 1;
+        this.hasNavigation = false;
+        this.hasBoundaryLinks = false;
+    }
+
+    // Public methods
+    public hasPrevious():boolean {
+        return this.page > 1;
+    }
+
+    public hasNext():boolean {
+        return this.page < this.pageCount;
+    }
+
+    public setPage(newPage:number):void {
+        const value:number = (Number.isInteger(newPage)) ? Math.min(Math.max(newPage, 1), this.pageCount) : 1;
```

Comment:
Good idea using `Number.isInteger`, though what are your thoughts on using `Math.round` instead? we'll need to add a null check beforehand though.